x-vector/src/X/Data/Vector/Generic.hs

```diff
@@ -14,12 +22,19 @@ module X.Data.Vector.Generic (
   , mapMaybeM
   , imapMaybeM
 
+  -- * Modifying vectors
+
+  -- ** Transposition
+  , transpose
+
```

Comment:
could you expose a slightly restricted version in `Data.Vector`, where the outermost is a Vector but innermost can be anything?
```
transpose :: forall va a. (G.Vector va a) => V.Vector (va a) -> V.Vector (va a)
```
it's just that I tend not to import `Vector.Generic` most of the time